### PR TITLE
Demo 2016

### PR DESCRIPTION
**Pull request by lzan13**
add sign out code

### CODE REVIEW

File: .idea/misc.xml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (githubzhaoliang) [inline .idea/misc.xml:1]: should not submit this file

File: app/src/main/java/com/hyphenate/chatuidemo/DemoHelper.java [hunk 2]
```diff
@@ -63,15 +109,119 @@ private EMOptions initChatOptions(){
...
+    public void signout(final EMCallBack callback) {
+        /**
+         * Call sdk sign out, this method requires two parameters
+         *
+         *  boolean: The first argument is required，Indicates that the push token is to be deallocated,
+         *  and if account offline, this parameter is set to false
+         *
+         *  callback: Optional parameter to receive the result of the logout
+         */
+        EMClient.getInstance().logout(isUnbuildToken, new EMCallBack() {
```
Comment (githubzhaoliang) [inline app/src/main/java/com/hyphenate/chatuidemo/DemoHelper.java:167]: isUnbuildToken  --> unbindToken

File: app/src/main/java/com/hyphenate/chatuidemo/DemoHelper.java [hunk 3]
```diff
@@ -63,15 +109,119 @@ private EMOptions initChatOptions(){
...
+                    callback.onProgress(i, s);
+                }
+            }
+        });
+    }
+
+    /**
+     * Check whether the login has been successful
+     */
+    public boolean isLoginedInBefore() {
```
Comment (githubzhaoliang) [inline app/src/main/java/com/hyphenate/chatuidemo/DemoHelper.java:193]: is this method necessary?

File: app/src/main/java/com/hyphenate/chatuidemo/DemoHelper.java [hunk 4]
```diff
@@ -63,15 +109,119 @@ private EMOptions initChatOptions(){
...
+        return EMClient.getInstance().isLoggedInBefore();
+    }
+
+    /**
+     * Determines whether the current app is connected to the chat server
+     *
+     * @return connection state
      */
-    private boolean isMainProcess(){
+    public boolean isConnection() {
```
Comment (githubzhaoliang) [inline app/src/main/java/com/hyphenate/chatuidemo/DemoHelper.java:202]: is this method necessary?
If so, use isConnected

File: app/src/main/java/com/hyphenate/chatuidemo/ui/settings/SettingsFragment.java [hunk 5]
```diff
@@ -1,30 +1,52 @@
...
+    @Override public View onCreateView(LayoutInflater inflater, ViewGroup container,
+            Bundle savedInstanceState) {
         // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.em_fragment_settings, container, false);
+        View view = inflater.inflate(R.layout.em_fragment_settings, container, false);
+        ButterKnife.bind(this, view);
+        return view;
     }
 
+    @OnClick(R.id.btn_sign_out) void singout() {
```
Comment (githubzhaoliang) [inline app/src/main/java/com/hyphenate/chatuidemo/ui/settings/SettingsFragment.java]: is this recommended code style?

File: app/src/main/java/com/hyphenate/chatuidemo/DemoHelper.java [hunk 6]
```diff
@@ -31,27 +49,41 @@ public synchronized static DemoHelper getInstance() {
...
+
         if(isMainProcess()) {
             //init hyphenate sdk with options
-            EMClient.getInstance().init(context, options);
+            EMClient.getInstance().init(context, initOptions());
         }
 
-    }
+        // set debug mode open:true, close:false
+        EMClient.getInstance().setDebugMode(true);
```
Comment (zhanw) [inline app/src/main/java/com/hyphenate/chatuidemo/DemoHelper.java]: 代码都写在isMainProcess()判断里面

File: app/src/main/java/com/hyphenate/chatuidemo/DemoHelper.java [hunk 7]
```diff
@@ -63,15 +95,119 @@ private EMOptions initChatOptions(){
...
+        };
+        EMClient.getInstance().addConnectionListener(mConnectionListener);
+    }
+
+    /**
+     * Sign out account
+     *
+     * @param callback to receive the result of the logout
+     */
+    public void signout(final EMCallBack callback) {
```
Comment (zhanw) [inline app/src/main/java/com/hyphenate/chatuidemo/DemoHelper.java]: signOut

File: app/src/main/java/com/hyphenate/chatuidemo/ui/settings/SettingsFragment.java (hunk 5 above)
Comment (zhanw) [inline app/src/main/java/com/hyphenate/chatuidemo/ui/settings/SettingsFragment.java]: signOut

**Pull request merged**